### PR TITLE
refactor restart_kraken and add tests, some refactor too

### PR DESCRIPTION
**Pull request by francois-vincent**
allow restart kraken to restart serially, ie no interruption of service

### CODE REVIEW

File: fabfile/component/kraken.py [hunk 1]
```diff
@@ -119,11 +119,10 @@ def require_monitor_kraken_started():
 
 
 @task
-def restart_all_krakens(wait=True):
+def restart_all_krakens(wait='serial'):
     """restart and test all kraken instances"""
-    wait = get_bool_from_cli(wait)
     execute(require_monitor_kraken_started)
-    for instance in env.instances.values():
+    for instance in env.instances.itervalues():
```
Comment (kinnou02) [inline fabfile/component/kraken.py]: ```itervalues``` is not python 3 compatible, we should keep ```values```

File: fabfile/component/kraken.py [hunk 2]
```diff
@@ -227,25 +226,38 @@ def check_dead_instances():
...
+               Possible values=False or None: restart in parallel, no test
+               'serial': restart serially and test
+               'parallel': restart in parallel and test
+        The default value is 'serial' because it is the safest scenario
+        to restart the krakens of an instance in production.
     """
+    if wait:
+        wait = wait.lower()
+        if wait not in ('serial', 'parallel'):
+            wait = False
```
Comment (kinnou02) [inline fabfile/component/kraken.py]: So wait can be a string or a bool?

Comment (francois-vincent) [inline fabfile/component/kraken.py]: it's a ternary value: serial, parallel or None===False

Comment (kinnou02) [inline fabfile/component/kraken.py]: I don't have any problems with None, but False is not None

**Pull request merged**